crates/engine/Cargo.toml

```diff
@@ -1,30 +1,32 @@
 [package]
 name = "ink_engine"
-version = "4.2.0"
-authors = ["Parity Technologies <admin@parity.io>", "Michael Müller <michi@parity.io>"]
-edition = "2021"
 
-license = "Apache-2.0"
-readme = "README.md"
-repository = "https://github.com/paritytech/ink"
-documentation = "https://docs.rs/ink_engine/"
-homepage = "https://www.parity.io/"
+edition.workspace = true
+categories.workspace = true
+homepage.workspace = true
+keywords.workspace = true
+license.workspace = true
+repository.workspace = true
+version.workspace = true
+
+authors = ["Parity Technologies <admin@parity.io>", "Michael Müller <michi@parity.io>"]
 description = "[ink!] Off-chain environment for testing."
-keywords = ["wasm", "parity", "webassembly", "blockchain", "edsl"]
-categories = ["no-std", "embedded"]
+documentation = "https://docs.rs/ink_engine/"
 include = ["Cargo.toml", "src/**/*.rs", "README.md", "LICENSE"]
+readme = "README.md"
 
 [dependencies]
-ink_primitives = { version = "4.2.0", path = "../../crates/primitives", default-features = false }
+derive_more = { workspace = true, features = ["from", "display"] }
+# cannot inherit due to: https://github.com/bkchr/proc-macro-crate/issues/34
 scale = { package = "parity-scale-codec", version = "3.4", default-features = false, features = ["derive"] }
```

Comment:
You can define this alias on the workspace level and use here `scale = { workspace = true }`

Reply:
please check my comment: https://github.com/paritytech/ink/pull/1835#issuecomment-1621570787

Reply:
Works for me:

<img width="995" alt="image" src="https://github.com/paritytech/ink/assets/18346821/4629c3ea-9b19-4603-aa75-3e87ef8bcb94">

Reply:
have you run the code? I have done it some time ago, but this doesn't compile: https://github.com/paritytech/ink/pull/1835/commits/08c19379d61b44e94dd08f36f09e0ffa86f6dcc0

(macro generation phase fails)

Reply:
btw, `package` is ignored when `workspace` is present

Reply:
https://github.com/pmikolajczyk41/ink/pull/1/files

Reply:
I will ensure that CI works well now and if so, I will apply these changes